src/components/ha-navigation-picker.ts

```diff
@@ -61,46 +23,53 @@ export class HaNavigationPicker extends LitElement {
 
   @property({ type: Boolean }) public required = false;
 
-  @state() private _opened = false;
-
-  private navigationItemsLoaded = false;
+  @state() private _loading = true;
 
-  private navigationItems: NavigationItem[] = DEFAULT_ITEMS;
+  protected firstUpdated() {
+    this._loadNavigationItems();
+  }
 
-  @query("ha-combo-box", true) private comboBox!: HaComboBox;
+  private _navigationItems?: PickerComboBoxItem[];
```

Comment:
You are using non null assertion in a couple of places but we can just initialize it here
```suggestion
  private _navigationItems: PickerComboBoxItem[] = [];
```